### ISSUE TITLE
I'm trying to reply to a post and get error(EventGhost forum problem)

### ISSUE THREAD

**Issue by jonisb** [EventGhost/EventGhost#417]
I'm trying to reply to this post [XBMC2 tread](http://eventghost.net/forum/viewtopic.php?f=9&t=1562&start=1440#p56525) but get this error:
![EventGhost forum error](https://user-images.githubusercontent.com/16629739/209487965-df8f57b3-95f5-4ba1-957e-fae305aaf779.png)

jonib

**Comment by jonisb** [comment]
Still getting this error. I don't see any new posts on the forum so don't know if it works for anyone else.
I tried with Pale moon and Edge browsers.

**Comment by ThomasBott** [comment]
I can confirm, same problem for me.. @kdschlosser: can you help?

**Comment by jonisb** [comment]
Thanks for confirming, I was helping a user and have no idea if his issue got resolved 😭

**Comment by jonisb** [comment]
Looks like http://[www.eventghost.net](http://www.eventghost.net/) is completely dead now. 🖖

**Comment by dequi** [comment]
The server isn't responding to ping's so I'm afraid it's down. I tought it was hosted by https://joesdatacenter.com/. 
At least the last one I had access to before the last server move (end march/2020?) was located there and I think Kevin stayed with them.
Kevin didn't respond to the mail I've sent him to his private address, but with the holidays and all it might take a while.
I've reached out to joesdatacenter and asked for a reboot if possible. Not sure if it's going to result in anything, as I am a stranger to them.

I've noticed the previous server is still up and I my access hasn't been revoked so I'm currently making backups of the data as it was on march/2020 - you never know.

**Comment by packetwarrior** [comment]
> The server isn't responding to ping's so I'm afraid it's down. I tought it was hosted by https://joesdatacenter.com/. At least the last one I had access to before the last server move (end march/2020?) was located there and I think Kevin stayed with them. Kevin didn't respond to the mail I've sent him to his private address, but with the holidays and all it might take a while. I've reached out to joesdatacenter and asked for a reboot if possible. Not sure if it's going to result in anything, as I am a stranger to them.
> 
> I've noticed the previous server is still up and I my access hasn't been revoked so I'm currently making backups of the data as it was on march/2020 - you never know.

Did you ever end up hearing back from Joe's Datacenter?  It looks like the current IP address eventghost.net resolves to is still allocated to them.

**Comment by dequi** [comment]
No - they did not respond yet. I'm currently Github stalking @kdschlosser for attention. I've posted a response to the last 2 issues he opened yesterday:  https://github.com/lvgl/lv_binding_micropython/issues/250 and  https://github.com/lvgl/lv_binding_micropython/issues/251 so: 🤞

**Comment by jonisb** [comment]
I'TS ALIVE or almost I was able to half load the EventGhost forum so it seems the server is not completely dead.
But refreshing to get it fully loaded it again died, no idea what this means, but if there is life there is hope 🙏

**Comment by dequi** [comment]
At this point I'm not that hopeful anymore: 
* I contacted https://joesdatacenter.com/ on two occasions but got no response, which as I am not a customer nor do I have the correct customer information is understandable. 
* I combed trough the last forum database backup that I have (March 2020): 
  * Found Kevin's gmail address: send two emails a couple of days apart - no response (yet?)
  * Found his wife's Denise hotmail address, sent an email - no response (yet?)
  * Found a phone number, called twice (different days) no pickup, has no voicemail configured. sent and sms, used the carriers email2phone service, sent a whatsapp - no response (yet?)
* Commented off-topic on 2 by Kevin recently opened Github issues, got no response but he responded on a later comment posted on one of these issues so I'm pretty sure he saw my post.
* Used the contact option on the eventghost.net whois info in order to contact the owner which I know for sure is Kevin - no response (yet?) - registration is expiring 18 Januari 2023 - *hope this does not elapse*

Then I went full "stalker" (for which I'm not proud), using the phone number I found:

* Confirmed this number is indeed registered to Kevin
* Found an accompanying home address - indeed in the Rockies as his github profile suggests
* Found the land line number for that address, called it, got message spoken by Denise, left a voicemail - no response (yet?)   

There is an eventghost slack channel but I deleted my account on that a couple of years back and I can't get back in. Maybe someone else could try to reach out to Kevin there?

I'm currently out of further ideas.

**Comment by dequi** [comment]
I may be grasping at straws but I've sent a mail to @Pako2 @CarsonF @topic2k and Zian in the hope @kdschlosser shared the server information with one of them, and we can get things running again.

**Comment by jonisb** [comment]
Thanks for your efforts dequi, hopefully this is not the end of EventGhost.

**Comment by CarsonF** [comment]
I'm happy to give anyone access to GitHub.
I don't have any idea about the website/forum. All of this is so old and unmaintained.
I'm not even sure what this offers anymore that Home Assistant doesn't.

**Comment by dequi** [comment]
@CarsonF could you possibly give access to jonisb and me, then we can start from there. If @jonisb could recommend one or two others then at least we are with three or four.

I got back here at the newyears because I was looking for automation software for some business related tasks on Windows pc and EventGhost was the first that came to mind as somewhat familiar. On the *home* end I use Node Red on a Synology NAS but I need something to be rooted in WIndows. Don't know why others are here but I'm sure they have their reasons.

**Comment by CarsonF** [comment]
> @CarsonF could you possibly give access to jonisb and me, then we can start from there.

Done.

I'd recommend migrating the website to host from GitHub to keep it here and simplify access and everything.

I'd recommend migrating the forum to GitHub discussions for the same reason.

**Comment by Ian-Zz** [comment]
This is Ian aka zian. First saw this thread last week or so when I noticed eg.net no longer loaded. I have zero recent knowledge or data concerning the site, domain, hosting etc. Have not seen KG post recently on FB. Have not spoke to him direct in maybe years.
Would so hate to lose all that shared knowledge over years and years. Will try to ping KG myself. Willing to do what it takes to save that forum and all its data.
Thanks to all working on the same.

**Comment by Ian-Zz** [comment]
No Shit. An Update.
I just checked my old seldom logged into email account and did a quick "evenghost" search. Came up with an email from support@domain-inc.net sent to me on 9-19-22.

It says in jist.... my details removed....
Dear Valued Customer,

ICANN, the organization responsible for the stability of the Internet,
requires that each domain name registrant be given the opportunity to
correct any inaccurate contact data (WHOIS data) associated with a domain
name registration. Our records for your domain are as follows:
eventghost.net

Domain Name: EVENTGHOST.NET
Registry Domain ID:
Registrar WHOIS Server: whois.domain.com
Registrar URL: www.domain.com
Updated Date: 2020-01-21T14:32:23
Creation Date: 2011-01-18T22:59:52
Registrar Registration Expiration Date: 2023-01-18T22:59:52
Reseller: Domain.com
Domain Status: ok https://icann.org/epp#ok
Registry Registrant ID:
Registrant Name: Ian 

I followed the links in the email and logged in to domain.com and have the chance to renew  the Domain Name: EVENTGHOST.NET for another year for $19.95 2 3 4 5YEARS FOR $99.95. Looks to me like we only have 3 days to ACT (at least on this domain name) portion of whatever the deal is with Eventghost.net going on now. Would just hate to pay $20 or $100 for something that isn't the main part of our problem.

Why is the site no longer loading? Anyone 100% certain why?
Thanks. Ian Feel free to email me. I'll hope to check it soon. er. Maybe.

**Comment by per1234** [comment]
I don't use EventGhost anymore, but I'll contribute towards the domain registration fee if needed.

**Comment by LoganAC34** [comment]
> I don't use EventGhost anymore, but I'll contribute towards the domain registration fee if needed.

I use event ghost all the time. I'll contribute too.

**Comment by jonisb** [comment]
@Ian-Zz Renewing the domain at least for a year should be fine, $20 Kickstarter lets goo 😸

**Comment by Ian-Zz** [comment]
Well whatever the problem is with the website we now have another year to fix it. At least in respect to the domain portion. I just purchased a years worth of all that's listed below at domain.com. 
eventghost.net
1 Year
 Domain $19.99 Renews 01/16/24
1 Year Privacy + Protection $8.99 Renews 01/16/24


1 Year Domain Expiration Protection $9.99 
Subtotal $38.97 Sales Tax$2.34
Total $41.31

**Comment by CarsonF** [comment]
@Ian-Zz you want to be an org owner here too?

**Comment by ThomasBott** [comment]
I'm also happy to support the project on git side; still using EventGhost at the office and for home automation and it's working great! I'm much to invested to change to another tool

**Comment by dequi** [comment]
> hopefully this is not the end of EventGhost.

I would say we are off life-support, and into intensive care I guess. Still lack of basic motor functions and big memory loss but still viable.

- [x] Get access to the EventGhost GitHub organisation/repository for multiple members - Thanks @CarsonF 
- [x] Get access to a EventGhost.net domain owner and get it renewed - Thanks @Ian-Zz
- [ ] Get access to the forum data - I only have the data until March 2020 - **missing almost 3 years worth of knowledge** 
I'm still hoping @kdschlosser responds back with the joesdatacenter info or a more more recent backup.
- [x] Get access to the old EventGhost website/docs data - I have these / can be regenerated
- [x] Get access to the old EventGhost wiki data - I have these

**Comment by dequi** [comment]
> I'm also happy to support the project on git side;

As you were quick to respond to jonisb 's post and already have contributed in the past. I deem you invested and worthy. :smile: 
I've sent you an invite. Now the burden is shared.

**Comment by Ian-Zz** [comment]
About that. We got pretty lucky with the domain situation. I had no clue that the domain info was still attached to me. I thought it was all changed over years ago at the time of the 2nd EG.net "remodeling".

That being said I think it would be very wise for me to "let go" of this domain and the account. Not sure how it works or if it's transferrable to another party (or two... or three). But seems the smart thing to do.

I know we went through this similar scenario years back. Who holds the keys. Who does and owns what. Who has access etc. But the way it's been the last few weeks (site down) and many viewers willing and able to do what needs to be down to get it back up again asap but looks to be be. One man gone = site down = nothing "we" can do. THAT SUCKS. Should be changed for in perpetuity. 

I know the bunch of you are very much mode familiar with all kinds of tech. Web, ESP, Python Etc Etc Etc. I'm not a coder and I for sure ain't the office keeping type. I shouldn't be in charge of shit. lol

Aren't there any tools or resources for running a website/forum/code base that is handled by a group or community of sorts. One that can also change and evolve over time. Always has many (or at least a few) competent controllers.

And thanks again to all of you chiming in and typing up. 'Specialy dequi and joni and others. I know you all spent years and lots and lots of time taking care of this project and it's users. For that we are all in debt. And that goes tripply for Kevin. The dudes been amazing. 'Till just recently. lol

Come on Kev. Where you at Bro?
Hope your well wherever you are.

**Comment by CarsonF** [comment]
Domains can be transferred, yes. I'd recommend creating a private org repo specifically for these org related secrets and private access info. DNS shared account login, etc. That way any member of the org can login and maintain. That still assumes all good actors obv. I'd recommend Cloudflare for standalone DNS.

**Comment by Ian-Zz** [comment]
CarsonF, Yes yes and yes.
I think I looked into .org last time back. Not sure why it was or wasn't done.

Again I'll say this. Back at the first EG change over 2011 I think. I was working with dns_forwards and names and such. Think I borked it in minutes the first few times. Same with the forum data. Close calls at least. Keep me away from any codes and/or any "high level registering" type shit. The forum anti-bot measures... That I could do. I think I was good at that one.

EG.net Forum... we miss you.
:(

**Comment by dequi** [comment]
A situation report:

At this moment I have hosting configured with the website / docs / forum as they were in March 2020. For the website and docs I confirmed with archive.org/waybackmachine that these are exactly as the latest version archived there (somewhere end 2022)

These are not accessible yet as DNS is not yet configured, but before that there is still and issue:

I've been thinking about the best way to moving forward with the forum is. 

  * How long do we wait on Kevin to potentially fix the forum to the state was in December? 
  * When re-starting without Kevin, what would be the better option:

**1. We start from 0 with GitHub Discussions**

* GitHub Discussions are fairly new and there are no good tools available for importing posts from a phpBB forum. So ALL history will be lost, instead of the last 3 years. Further: API's for exporting / backup are not available as far as I could find.  

* Everyone needs to create a GitHub account to be able to post. With the potential of someone with bad intentions registering with a username of a "trusted" person. 

* In order to preserve some of the past we could host the phpbb forum read-only and locked off. But I fear it would create a more fragmented "getting help" experience. 

**2. We continue the posting on top of the backup of March 2020** 

* *What happens if a more recent forum backup shows up?* 

Merging 2 phpBB's has been done before but it is a big pain in the \*\*s and success rates are low. 
There are 37 tables with an "id" field that uses an AUTO_INCREMENT value.
I've determined that 10 of those will make it nearly impossible to merge data back in from a more recent backup (if it ever shows up), because we will be stuck with dupicate id's that have different data attached. A solution to improve future chance of merge success is to increase the next AUTO_INCREMENT value of those tables to double what it was, before restarting, thus preserving "id" space between the time the backup was made and  when we now continue. 

* *What about users who registered between march 2020 and now?*

These users would have to re-register. It's like they never existed.
But here is the kicker! When a more recent backup does appear, and we decide to merge it in, it would result in dupicate usernames: one with and old and one with a new id. Theoretically these could be merged but I fear this would be difficult. As a solution we could just then prepend the oldest identical username with something like "*historic_*" and leave it at that. 

I'm leaning to the second option, but what do you guys think? any other suggestions/options or pitfalls I'm missing.

**Comment by LoganAC34** [comment]
Just throwing in my 2 cents; I vote for github and hosting the old forum as read only, though I don't really understand everything involved as im am an amateur here, so take my opinion with a gain of salt. 

To me, it seems like using github would be better since it would negate all the issues you mentioned in option 2 which seems tough to deal with. You could still merg any new data you get from Kevin later if the site remains untouched. People can just link to the old forum in github discussions to either reference or continue a post. To help cut down on duplicate posts, you could note in github posting rules to search the old site first. It also leaves open the possibility of importing the forum data into github should we want to do that. It also keeps things together on one site (excluding old forum). Also, I always prefer github UI over other forums, though that's just from a user's perspective and not an admin. 

I can't say I understand fully what you mean with "...someone with bad intentions registering with a username of a "trusted" person.", but it seem like that could be handled. I don't think creating a github account is a problem.

Again, this is the opinion of an amateur, so I may not know what I'm talking about. Just wanted to give my thoughts.

**Comment by jonisb** [comment]
Hi @dequi maybe you should create a new issue (and pin it) and move this discussion there, as other people might not see this important discussion.
If I have time tomorrow I'll check the report properly and comment on it.

**Comment by packetwarrior** [comment]
As someone that has been running forums for over 20 years, I'd go with option 1 as well.  In fact, I'd go as far as not bothering with a public read-only copy of the old forums as now you're responsible for maintaining software, hosting and security.  We also don't know why the site broke this time.  It could have been a security incident or scare, and Kevin decided it was just safer to turn it all off vs having the data leak.

We can always rebuild some of the critical threads if needed, but we lost a lot of data several years ago, so I don't think it's going to be too difficult of a change.

**Comment by siparker** [comment]
just in case it helps. there is quite a bit of info on archive.org for Eventghost. including the forum. some pages as recently as December 22
[https://web.archive.org/web/20221128185249/http://www.eventghost.net/forum/index.php?sid=4146accecd08a51243783531ddbd2d82](https://web.archive.org/web/20221128185249/http://www.eventghost.net/forum/index.php?sid=4146accecd08a51243783531ddbd2d82)

**Comment by Ian-Zz** [comment]
> just in case it helps. there is quite a bit of info on archive.org for Eventghost. including the forum. some pages as recently as December 22 https://web.archive.org/web/20221128185249/http://www.eventghost.net/forum/index.php?sid=4146accecd08a51243783531ddbd2d82

Very cool...
I just looked into this. At the moment my plate is full.
Plan C maybe. ;)
https://superuser.com/questions/828907/how-to-download-a-website-from-the-archive-org-wayback-machine

**Comment by blackwind** [comment]
Thank you, Ian, for finally stepping up and doing the right thing. It's unfortunate that we're being dragged kicking and screaming to where I would've liked to have seen things go many, many years ago, but this will all ultimately be for the best once the dust settles.

I concur with Carson's recommendations here. Logan's assessment of the forum options is spot-on. A non-profit organization should be registered and domain ownership (the contact details) should be transferred to it. Determine if Kevin is in or out and remove him from the GitHub organization if needed before publishing any private secrets. If he's actively ignoring us in other threads as dequi suggests and not completely incommunicado due to, for example, a potential temporary medical issue, I think that tells the tale. Migrate the domain to Cloudflare, publish the Cloudflare account details privately, attach the domain to GitHub Pages, and move forward. 👍

**Comment by ntzb** [comment]
I agree with @LoganAC34 view of things.
github issues going forward, and hosting the old forum as read only (losing all of the past topics and knowledge could be a deadly blow).
I will happily chip in on costs

**Comment by getlucky3872** [comment]
likewise, happy to chip in as well

**Comment by Soupy51** [comment]
I had a serious injury a while back and have been bedridden for quite awhile. I have been using the web interface t.bott has developed.  My skills as a Python, JavaScript programmer et al are rudimentary. But I have money to help.  Please don't hesitate to forward any financial plans my way.  Your non-profit idea sounds workable.

**Comment by gorman42** [comment]
I am no coder, I am no web developer. But I use EG daily and I think it's a damn shame that all the forum knowledge is lost. If there's a money contribution that could go towards something useful, I'd be in.

**Comment by Soupy51** [comment]
What is the current ststus of forum backups? Where are they, who has them.
Can we set up a new forum for a place to discuss and start anew regardless of backup data status.  Website status? Who has access?  Can we please start with getting the website back up?  Can someone volunteer who has html experience and time?  Please chime up with direction?

**Comment by ZeroPain** [comment]
i would so donate to get eventghost going again nothing compares

**Comment by krambriw** [comment]
So sad to read all this. I was wondering myself. During the years since I left the community I have had so many people asking for links to my plugins. I tried to ask Kevin to update them on the forum but no response. SInce a while back, those requests have stopped coming, which feels good in some way, now I understand why

As you all know, I was earlier very active on the forum. After I left the community I started instead with Node-RED. Even though Node-RED is based on javascript and running on node.js I am rather impressed. I have migrated all functionality I need and also been able to re-use many of my Python babies, interfacing them using MQTT. With all your skills I think you should be able to migrate whatever you have in EG to Node-RED, so many plugins (called nodes) are available, also for all kind of devices. An extremly & helpful community there as well, like the good old days we had

Check it out: [https://discourse.nodered.org/](url)
 
Kind regards, Walter

**Comment by steffenu** [comment]
Sad to say now chrome webstore will take down a plugin of mine related to eventghost 

this is the end.

https://chrome.google.com/webstore/detail/browser-remote-for-eventg/fogmdbdpboenfkpamnfceijhbjhoablm?hl=de

farewell

**Comment by ntzb** [comment]
> I have migrated all functionality I need and also been able to re-use many of my Python babies, interfacing them using MQTT. With all your skills I think you should be able to migrate whatever you have in EG to Node-RED, so many plugins (called nodes) are available, also for all kind of devices. An extremly & helpful community there as well, like the good old days we had

I gather node-red is not a drop-in replacement, but can it actually support all of the functionality that EG had?
I mean many events were coming from Windows APIs etc..

**Comment by krambriw** [comment]
I do not have a one-to-one comparison. Both are event handling machines. It all depends a lot on your personal requirements. I'm not saying that it will come without some work, even some sweat & tears might be needed, but there is a huge base of supported devices and api's to all kind of hardware and sofware. You have to diggin to get a basic undrstanding and overview how it can fit you needs.

Remember, many of the Windows API's you might refer to was just needed for EG itself. Node-RED runs on Windows as well as on Linux and basic programming is visual but you still have coding (in javascript) there when needed 

After using it for several years, I think I have a more advanced and sophisticated system today with many more options and possibilities than before with EG. Everything I could do and did in EG, I have now in Node-RED. I migrated step by step while I had EG still running my operational system. Once done I did put EG to rest.

Node-RED is future proofed, it is backed by JS Foundation and a huge active & extremly helpful community. So whatever functionality you need, just make a search with the additional keyword "Node-RED" and most likely you will find a solution. When you decide to take the step, you will not regret and go back

**Comment by dequi** [comment]
> just in case it helps. there is quite a bit of info on archive.org for Eventghost. including the forum. some pages as recently as December 22 https://web.archive.org/web/20221128185249/http://www.eventghost.net/forum/index.php?sid=4146accecd08a51243783531ddbd2d82

Just spent the last couple of days scraping the data available on archive.org but the result is disappointing; The archive.org crawls are very shallow and only about 26% of the last 3 years is (partially) recoverable. I've also noticed that Bing claims to have about [4530 forum results](https://www.bing.com/search?q=site%3Aeventghost.net%2Fforum) "cached" but after result page 8 it just errors out. Google doesn't return any results that might be cached anymore. 

I don't expect @kdschlosser has made backups otherwise he would have chimed in already. Maybe by some miracle someone still has some cache in his or her browser that they can extract, or has done a scrape by the end of December 2022 that they can share, but for me it ends here. I don't have the time available to rebuild this community basically from scratch.    

As others have mentioned before, also I have moved on to other solutions.

**Comment by Soupy51** [comment]
Nah, I'm just needing plug-in logic to hurry along the transfer.  I got all
that right here.  Thanks again!

Steve C

**Comment by Soupy51** [comment]
Watch a tutorial on utube then follow the instructions on website. Thats as
far as ive gotten

**Comment by Soupy51** [comment]
Yup.  Just start with youtube.  Ton of info.

**Comment by krambriw** [comment]
And then join the forum at discourse as mentioned above, we all see each other there (my user name is the same). Very helpful and most professional users are there to help with all questions and solutionsCheck it out: [https://discourse.nodered.org/](https://github.com/EventGhost/EventGhost/issues/url)

**Comment by Soupy51** [comment]
Thanks for the tip!

**Comment by atrugiel69** [comment]
I go to nodered.org, search for "windows events" comes up empty.

There are nodes for Kodi, both directions (sending commands, receiving events) but I can't find the same for Windows. Am I missing something?

**Comment by krambriw** [comment]
Yes, Node-RED is not only for Windows, what specifically are you looking for?

**Comment by atrugiel69** [comment]
> Yes, Node-RED is not only for Windows, what specifically are you looking for?
> 
> https://flows.nodered.org/search?term=events

Yeah, I did that search but there's the problem, right there.

EventGhost had a portion of events listening "built in", as far as I can tell, all those produced by Windows OS. Is there a way to access those through Node-RED? Because extending what's been listened to through plugins (nodes in this case) is familiar (say for Kodi), but I'm not sure how to proceed for Windows itself.

**Comment by krambriw** [comment]
I'm sure there is, you have even better possibilities & flexibility. Exactly these type of questions are perfect to put on the forum and a lot of experts and experienced users will welcome you. One thing is most of the time true, if you are looking for something, most likely others have done the same and might already have a solution. The more detailed you can be, the best chance to get a good & useful response

**Comment by wysocki** [comment]
Node-Red appears to be HUGELY more difficult to use than EG. Non-starter in my book.

**Comment by Soupy51** [comment]
Sorry to hear that.  I am a guy who gets the seniors discount at the local
Denny’s.  Took me about 20 hours to get my tplink tapo equipment working.
Just waiting on a new camera, experimenting with a wifi pan/tilt camera.
According to forum, easy to get going.  I'm sticking with it...the
investment in time seems to be worth it.

**Comment by krambriw** [comment]
> Node-Red appears to be HUGELY more difficult to use than EG

Difficult no, different yes. But it will depend on what your use cases are. Remember EG was originally developed mainly to integrate video/audio/remote control stuff and then expanded in various directions. So if it is that you are looking for, yes EG had such stuff out-of-the-box where Node-RED is a more generic framework with thousands of add-ons, some suppporting video/audio equipment. Even supported by some manufacturers like Samsung. Never saw that happen with EG
https://flows.nodered.org/node/node-red-contrib-samsung-automation-studio-nodes

**Comment by DanEdens** [comment]
I have a dream of rewriting eventghost to use mojo. Anyone interested?

**Comment by gorman42** [comment]
> I have a dream of rewriting eventghost to use mojo. Anyone interested?

Anything that could bring EventGhost back in action would me most welcome! But I'm sadly no developer...

**Comment by wysocki** [comment]
Anything to bring back EG! Actually Mojo appears to be a great new language that combines the usability of Python with the performance of C. Since it's so similar to Python, wouldn't it be possible to migrate at least some parts of EG into it? And since Python is an interpretive language wouldn't its source code be easily available (or decompiled)? I'm an old programmer so I may have already lost my mojo, but this would be a very interesting approach (I'd at least help fund it)!

**Comment by Telorast** [comment]
Looking for an alternative to EG, seems like there are lots with good hardware support but how do I replace the integrations with Windows software? That seems to be missing from Node-Red, Home Assistant, and others I have looked at. Am I missing something or how do I interact with Windows and other software without EventGhost?

For example I have shortcuts on my phone that sends messages through Tasker with AutoRemote to EG and then on to other programs like DisplayFusion, MediaMonkey, Perfect Keyboard, VoxCommando or various Windows commands like volume control, sound output or putting monitors in standby.

**Comment by Soupy51** [comment]
Please, By all means ask that question in NodeRed forums. You will find
your answer no problem!

s

**Comment by DanEdens** [comment]
There is no comparison to eventghost for windows. I have searched and searched, and nothing is as good, except maybe paid products like UIpath. I will start trying to get the mojo revival going, @kdschlosser if you want to chime in, I'd love that. but Cheers I get it xD

**Comment by gorman42** [comment]
> There is no comparison to eventghost for windows. I have searched and searched, and nothing is as good, except maybe paid products like UIpath. I will start trying to get the mojo revival going, @kdschlosser if you want to chime in, I'd love that. but Cheers I get it xD

I am no developer but if there is any way for me to help (testing, I don't know what else), I'd be down for it.

If this gets serious and you would accept donations, I would surely be ready to donate for the cause.

**Comment by DanEdens** [comment]
I can't actually begin work on this until the Windows version of Mojo is released, But This is the new fork I will be attempting to convert Eventghost from stackless 2.7 into Mojo. I will also need to see what incapability problems windows 11 will bring, This is going to be a learning experience for me, but I am determined to get this building, and then continue to develop it into something greater. If we can get it to build on mojo, then I will try to start a new forum and get more people involved. 

https://github.com/DanEdens/EventGhost-Mojo

**Comment by chip33** [comment]
@DanEdens 
In case you dont have it here is the patch that fixes issues in the last release of EventGhost 0.5.0-rc6.

 [EventGhost05rc6patch.zip](https://github.com/EventGhost/EventGhost/files/12566521/EventGhost05rc6patch.zip)

**Comment by DanEdens** [comment]
Awesome thank you!! 

Not sure if I have it or not so I'll compare.
 When did this happen? I'm using one of Kevin's WIP versions with some re-themeing 
😄
(link just has screenshot, post is about restarting this) 
https://www.linkedin.com/posts/danedens_mojo-eventghost-activity-7106076792158842880-yJma?utm_source=share&utm_medium=member_android

**Comment by chip33** [comment]
You are welcome! 
I can follow your fork on github.
I believe the patch is at least 2 years old but definitely fixes issues that were in rc6. 
I was active on the forum for a while 2 years ago and this patch was made as it was difficult to get changes put into github and released officially. I was chatting with the man who made it and he was developing his own plugins as well.
You can download a copy of what remains of the forum using this software.
[https://github.com/hartator/wayback-machine-downloader#installation](url)
It comes in at 1.43GB.

I did download a copy  a month or so ago but had not looked at it. I just tried to open the forum section and it is blank just using Chrome.
I also have a plugins folder here that contains as many plugins as I could collect at the time.
[plugins.zip](https://github.com/EventGhost/EventGhost/files/12566553/plugins.zip)

**Comment by DanEdens** [comment]
I'll check it out and return any I have that you don't! 
The loss of that forum is seriously a great loss. 
It was chalk full of the most complicated and educational threads I've ever seen, It's what got me really into Python.😁
Thanks so much for sharing these

**Comment by gorman42** [comment]
> I can't actually begin work on this until the Windows version of Mojo is released, But This is the new fork I will be attempting to convert Eventghost from stackless 2.7 into Mojo. I will also need to see what incapability problems windows 11 will bring, This is going to be a learning experience for me, but I am determined to get this building, and then continue to develop it into something greater. If we can get it to build on mojo, then I will try to start a new forum and get more people involved.
> 
> https://github.com/DanEdens/EventGhost-Mojo

I can't set up a build environment but I will follow development and I am ready to test whatever you'll want to throw my way.

Thanks for providing a glimmer of hope for a future with EventGhost.

**Comment by chip33** [comment]
@DanEdens 
>  I'm using one of Kevin's WIP versions with some re-themeing  <

Hi, would it be possible to share the code changes used to create the dark theme UI you show in the image in your post above?

**Comment by DanEdens** [comment]
Okay so bare with me a little and I'll try to fix it after I explain.

So this was one of the VERY FIRST things I ever learned how to do in python. It was that AHA moment that really turned me into a Python dev. Eventghost was completely open for me to chop to pieces! 

But it's sloppy :D Very sloppy. So sloppy that I still have hardcoded hacky paths in my personal copy that won't work for your machine. I have custom menu options and some other weird stuff that wouldn't make sense lol. 

That being said. Checkout this commit and I think we can walk you through how to edit it yourself. 

https://github.com/DanEdens/EventGhost/commit/4116e4d6cdc5622a3560f190c5433dd8c4449109

**Comment by DanEdens** [comment]
@chip33 Were you able to play with the colors?

**Comment by chip33** [comment]
Hi Dan, 
I swapped out the original RGB values for the new ones in all 3 files 
[eg/Classes/MonitorsCtrl.py]
[plugins/System/__init__.py]
[plugins/Webserver/__init__.py]
I figured the other changes were just related to the other things you do in that commit.

However just replacing this line in [plugins/System/__init__.py]
 `self.SetBackgroundColour(wx.Colour(0, 0, 0))`
with 
`self.SetBackgroundColour(wx.Colour(195, 195, 195))`

causes the following rush of errors in the log so I just left it as I couldn't think of an obvious reason the change in one set of RGB values would cause this..
19:24:48   ---> Welcome to EventGhost <---
19:24:48   Traceback (most recent call last) (0.5.0-rc6):
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ActionThread.py", line 80, in StartSession
19:24:48       pluginInfo = eg.pluginManager.OpenPlugin(guid, None, ())
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\PluginManager.py", line 52, in OpenPlugin
19:24:48       clsInfo = NonexistentPluginInfo(ident, evalName)
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\PluginManager.py", line 140, in __init__
19:24:48       Plugin.__name__ = name
19:24:48   TypeError: can only assign string to Plugin.__name__, not 'NoneType'
19:24:48   Unhandled exception in WorkerThread <EventThread>:
19:24:48   Callers stack:
19:24:48     File "wx\_core.pyc", line 8657, in MainLoop
19:24:48     File "wx\_core.pyc", line 7952, in MainLoop
19:24:48     File "wx\_core.pyc", line 16766, in <lambda>
19:24:48   Traceback (most recent call last) (0.5.0-rc6):
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ThreadWorker.py", line 253, in __DoOneEvent
19:24:48       self.HandleAction(action)
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ThreadWorker.py", line 140, in HandleAction
19:24:48       action()
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ThreadWorker.py", line 324, in __call__
19:24:48       self.returnValue = self.func(*self.args, **self.kwargs)
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\EventThread.py", line 75, in StartSession
19:24:48       actionThread.Func(actionThread.StartSession, 120)(filename)
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ThreadWorker.py", line 324, in __call__
19:24:48       self.returnValue = self.func(*self.args, **self.kwargs)
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ActionThread.py", line 87, in StartSession
19:24:48       eg.document.Load(filename)
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\Document.py", line 339, in Load
19:24:48       root = self.RootItem(self, node)
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\RootItem.py", line 48, in __init__
19:24:48       ContainerItem.__init__(self, parent, node)
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ContainerItem.py", line 34, in __init__
19:24:48       for childNode in node
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ContainerItem.py", line 34, in __init__
19:24:48       for childNode in node
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ContainerItem.py", line 34, in __init__
19:24:48       for childNode in node
19:24:48     File "C:\Program Files (x86)\EventGhost\eg\Classes\ActionItem.py", line 60, in __init__
19:24:48       plugin = getattr(eg.plugins, pluginStr).plugin
19:24:48   AttributeError: 'Bunch' object has no attribute 'System'`

**Comment by SamuelWest** [comment]
Hi @DanEdens I'm a reasonably experienced python dev have written a couple of EG plugins. There's nothing as good as EG for windows, so it'd be terrible if it died! Is there anything in particular that I could help with?

**Comment by ThomasBott** [comment]
Basically EventGhost needs to be upgraded to python 3 to have a future..

**Comment by steffenu** [comment]
Also created a few Eventghost Plugins back in the day. So im Hoping to see the legacy of Eventghost continue. That being said . I think that will be difficult as it is right now. This Project is probably dead and wont be revived.  My EG Chrome Extension were killed and burned along with it by google as the site went down :D.

Updating Eg to Python 3 wont work for a future. Since all plugins are Python 2 based. You could update the core easily. But it would be a empty python 3 shell without any plugins. So just by that you see the vastness of effort required.

At this point it makes sense to make something  new. But that requires a good amount of experience. And would be a huge undertaking. Maybe further down the road. I have professional Webdev Experience , long time hobby pythonista and  going into State of the Art Software next year. 

Im Interisted to see if there are still people willing to dedicate time to this or more likely a new project.
Feel Free to send me  mail if you got the passion , time and willingness to pull smth off together. Would be happy to chat about it. Miss the old good days with @DanEdens , @kgscchlosser.

**Comment by ThomasBott** [comment]
Of course the plugins would have to be adjusted as well but without the core running python 3 no plugin dev would update their plugins. Modifying a plugin to work with Python 3 is also no big deal - Someone has to start, I guess :)

**Comment by chip33** [comment]
Updating the plugins to work with Python 3, isn't this just the kind of thing that Chat GPT is good for? It can make mistakes that will need checking but could make it a lot easier.

>M odifying a plugin to work with Python 3 is also no big deal

**Comment by Telorast** [comment]
With so much lost beyond recovery, a python 3 reboot sound like a good idea. Even with a much reduced plug-in list EventGhost would be an attractive option with some active developers.

Considering how many other options there are for creating the rules behind a smart home now it might make sense to focus more on being the Windows node, passing on events and acting on commands from another system.

**Comment by SamRWest** [comment]
@DanEdens, @kdschlosser how functional is https://github.com/kdschlosser/EventGhost-x64-Python3.5 at the moment?

After some fiddling with dependencies and installing lots of MS stuff I can build and kind of run it locally, but get a bunch of runtime errors which I _think_ are from some python 2 syntax in the `eg` packages that are still yet to be translated. 

Just wanted to understand whether I'm doing something wrong with the build (entirely possible, NFI how the build script works yet), or if it's actually just not yet functional?

If the latter, any idea how much effort would be required, and is this code a good starting point?

**Comment by MP-Munich** [comment]
Is it planned to bring the EventGhost website and forum back online?

I'd love seeing this fabulous application still alive! 😊

**Comment by steffenu** [comment]
> Is it planned to bring the EventGhost website and forum back online?
> 
> I'd love seeing this fabulous application still alive! 😊

Talked with @DanEdens about this. 
I will probably will start create somthing fresh from the ground up . Maybe  there will be an announcement this year.  Stay tuned :)

**Comment by asheroto** [comment]
Site is totally down now...

**Comment by automationghost** [comment]
> Site is totally down now...

im 2 monthts in on developing a new project.
Its going pretty well and im going to stuff some cool features into this 

I expect to release a version this year. Maybe the website will be up a little sooner so i could get some feature request / alpha testers and so on :) ...

stay tuned :) . Regarding all the lost plugins ...  Maybe we can adapt some of these back into the new app with the help of the community . I will have python 2 , 3 support , as well as node , typescript and C# which you can use for running automations.  

Greetings

**Comment by jibblet** [comment]
I would love to help in any way possible - it's crazy to me that EventGhost got abandoned. I'm Python, webdev, etc.

**Comment by asheroto** [comment]
If you need a free spot to host a landing page or at least redirect to the EventGhost repo, GitHub pages works. Right now tons of links online are dead because the site is down. :-/ 😊

**Comment by automationghost** [comment]
For anyone reading this : feel free to join the automationghost discord channel :
https://discord.gg/TTvnSyPV

Big help would be testing the app when I release  a test / alpha version . 
Could share some early screenshots / ideas there so we can talk about what can be added etc.. 
I will reward testers with free version :) .

**Comment by jovrstra** [comment]
Long time EventGhost user here, been monitoring this repository ever since the site went down... I tried to replace EventGhost, but nothing even comes close to it...

I tried to sign up to the discord channel, but the link expired... Any chance to still join in? If I can I would love to help with beta testing.

**Comment by automationghost** [comment]
the link shows still as active .. created a new one just in case

https://discord.gg/bx2WAvW3

**Comment by jovrstra** [comment]
It worked! Thanks

**Comment by K4in-5037** [comment]
> Considering how many other options there are for creating the rules behind a smart home now it might make sense to focus more on being the Windows node, passing on events and acting on commands from another system.

This is how and why I and still use eventghost, for 2 way communication between Windows an Home Assistant, using the mqtt plugin I automate actions based on events in windows and from devices connected in HA to Windows.

**Comment by asheroto** [comment]
If the domain isn't renewed and we can't maintain contact. I'll plan on purchasing it when it's dropped to avoid scammers getting ahold of it.

**Comment by DanEdens** [comment]
I've started a Rust rewrite to get around the python3 issue. 

We might be able to return to python once GIL is actually able to be removed, But I've made good progress documenting the structure and Believe I can produce something nearly identical in Rust. (I tried doing this in Mojo already, but I didn't really like Mojo.)

Follow it here.[ ](https://github.com/DanEdens/EventGhost-Rust)

I've reached out to @kdschlosser but No luck so far. 
Hoping I can get added to the project, and republish as Eventghost/Eventghost-Rust or something.

**Comment by gorman42** [comment]
> Guys stay tuned I am still working on automationghost brand new project
> 
> Expected Beta Release early 2025. Its a new concept that will make use of existing smarthome integrations like HomeAssistant instead of relying on dusted / abandoned plugins that never recieve updates :).
> 
> https://www.automationghost.com/ ( This is just a draft / placeholder for now )
> 
> There is also a link to join discord group if you scroll up.

And this would be compatible with existing plugins and configurations?

**Comment by automationghost** [comment]
> > Guys stay tuned I am still working on automationghost brand new project
> > Expected Beta Release early 2025. Its a new concept that will make use of existing smarthome integrations like HomeAssistant instead of relying on dusted / abandoned plugins that never recieve updates :).
> > https://www.automationghost.com/ ( This is just a draft / placeholder for now )
> > There is also a link to join discord group if you scroll up.
> 
> And this would be compatible with existing plugins and configurations?

No plugin support for EventGhost is planned. Most of the history is gone anyway,  
But you can keep both apps running at the same time no problem and transition

**Comment by gorman42** [comment]
Sorry quoted the wrong message.

**Comment by gorman42** [comment]
> I've started a Rust rewrite to get around the python3 issue.
> 
> We might be able to return to python once GIL is actually able to be removed, But I've made good progress documenting the structure and Believe I can produce something nearly identical in Rust. (I tried doing this in Mojo already, but I didn't really like Mojo.)
> 
> [Follow it here. ](https://github.com/DanEdens/EventGhost-Rust)
> 
> I've reached out to @kdschlosser but No luck so far. Hoping I can get added to the project, and republish as Eventghost/Eventghost-Rust or something.

And this would be compatible with existing plugins and configurations?

**Comment by dequi** [comment]
> If the domain isn't renewed and we can't maintain contact with the developer, ~I'll plan on purchasing the domain when it's dropped to avoid scammers getting ahold of it. Then I will redirect it to the repo.~ We may need to create a separate repo if no one has full access.
> 
> Note, just read this comment: [#417 (comment)](https://github.com/EventGhost/EventGhost/issues/417#issuecomment-1384780399)

Currently there are 8 owners on the EventGhost repository but only Kevin is listed as public. I'm one of them but I'm currently lacking the time to contribute much. I don't have access to the domain but I've reached out to @Ian-Zz to see if he will renew it. It expires the 18th of January

**Comment by asheroto** [comment]
Sounds good, thank you.

**Comment by atrugiel69** [comment]
> > hopefully this is not the end of EventGhost.
> 
> I would say we are off life-support, and into intensive care I guess. Still lack of basic motor functions and big memory loss but still viable.
> 
>     * [x]  Get access to the EventGhost GitHub organisation/repository for multiple members - Thanks @CarsonF
> 
>     * [x]  Get access to a EventGhost.net domain owner and get it renewed - Thanks @Ian-Zz
> 
>     * [ ]  Get access to the forum data - I only have the data until March 2020 - **missing almost 3 years worth of knowledge**
>       I'm still hoping @kdschlosser responds back with the joesdatacenter info or a more more recent backup.
> 
>     * [x]  Get access to the old EventGhost website/docs data - I have these / can be regenerated
> 
>     * [x]  Get access to the old EventGhost wiki data - I have these

Missing forum data is downright criminal. I can understand life taking precedence over development of a free program, totally. Completely. But the data that was the fruit of hundreds of users' labour... that is a crying shame to let it disappear.

**Comment by Ian-Zz** [comment]
I have no access to database. But concerning the eventghost.net domain at domain.com that was in my name. Months back I removed my payment info from it and also stopped auto renew. Just checked at domain com and here is the stats... eventghost.net
Was protected from expiring but requires renewal
Domain Expiration Protection (eventghost.net)
Expired
01/18/2024
Domain Privacy + Protection (eventghost.net)
Expires Soon
01/16/2025

If anybody wants it transfered to them. We have a few days it looks like. I know before when I tried to share the eg.net domain.com admin privileges with others they made it damn near impossible. I am willing to do what's needed to hand it over.

Ian

**Comment by wysocki** [comment]
@Ian-Zz: I'm not a developer and certainly have no skills that can redevelop EG (although I'd have a hard time living without it). But if you are only in need of someone to take charge of the domain, I suggest you hold on for a while and I'd be very happy to pay the renewal fees for it. Certainly don't let it expire soon. What are we, the US congress???

Also @atrugiel69 says "...I can understand life taking precedence over development..."
Maybe I missed it, but seems to me that no one has openly acknowledged the situation here: death has taken its precedence.

I apologize to anyone offended by this posit, but Kevin Schlosser has apparently vanished from the face of the earth. Please convince me that I'm wrong. How can there be any other explanation?

**Comment by gorman42** [comment]
> Also @atrugiel69 says "...I can understand life taking precedence over development..." Maybe I missed it, but seems to me that no one has openly acknowledged the situation here: death has taken its precedence.
> 
Oh... Never thought about that. I'd be truly sorry.

**Comment by per1234** [comment]
>Please convince me that I'm wrong. 

You only need to look at their GitHub profile to see that Kevin is still very much alive and kicking:

https://github.com/kdschlosser

![image](https://github.com/user-attachments/assets/5b1cd358-abae-42d5-b54b-5d26173786cb)

**Comment by per1234** [comment]
> I'd be very happy to pay the renewal fees for it.

My previous offer to contribute to the domain registration fees still stands.

**Comment by asheroto** [comment]
We are discussing this with Kevin and a few others on a separate issue, more to come soon. 😊 The domain is being transferred.

**Comment by Ian-Zz** [comment]
> @Ian-Zz: I'm not a developer and certainly have no skills that can redevelop EG (although I'd have a hard time living without it). But if you are only in need of someone to take charge of the domain, I suggest you hold on for a while and I'd be very happy to pay the renewal fees for it. Certainly don't let it expire soon. What are we, the US congress???
> 
> Also @atrugiel69 says "...I can understand life taking precedence over development..." Maybe I missed it, but seems to me that no one has openly acknowledged the situation here: death has taken its precedence.
> 
> I apologize to anyone offended by this posit, but Kevin Schlosser has apparently vanished from the face of the earth. Please convince me that I'm wrong. How can there be any other explanation?

Yeah. The way this went down is pretty FUCKED UP.

**Comment by atrugiel69** [comment]
> We are discussing this with Kevin and a few others on a separate issue, more to come soon. 😊 The domain is being transferred.

What is the situation with forum's content. It's incredible to think about the amount of knowledge that went down the drain.

**Comment by jibblet** [comment]
> > We are discussing this with Kevin and a few others on a separate issue, more to come soon. 😊 The domain is being transferred.
> 
> What is the situation with forum's content. It's incredible to think about the amount of knowledge that went down the drain.

great point. I can't count how many times I googled something EG-related and got a link to the defunct forum.

**Comment by asheroto** [comment]
> What is the situation with forum's content. It's incredible to think about the amount of knowledge that went down the drain. 

I was told that there was an issue at the data center with the web server that the forum was hosted on, so the forum data was sadly lost.

Some possible good news, though... Kevin mentioned that he may have a backup of the forum database. He's checking to see if this is still the case. If so, I'll rebuild the forum using the backup. I'll keep you posted. 😊

**Comment by jibblet** [comment]
> Some possible good news, though... Kevin mentioned that he may have a backup of the forum database. He's checking to see if this is still the case. If so, I'll rebuild the forum using the backup. I'll keep you posted. 😊

Thank you!

Out of curiosity - is there a reason why Kevin has stepped away from (and all but abdicated, tbh) any role in this transition?

**Comment by dequi** [comment]
> I have no access to database. But concerning the eventghost.net domain at domain.com that was in my name. Months back I removed my payment info from it and also stopped auto renew. Just checked at domain com and here is the stats... eventghost.net Was protected from expiring but requires renewal Domain Expiration Protection (eventghost.net) Expired 01/18/2024 Domain Privacy + Protection (eventghost.net) Expires Soon 01/16/2025
> 
> If anybody wants it transfered to them. We have a few days it looks like. I know before when I tried to share the eg.net domain.com admin privileges with others they made it damn near impossible. I am willing to do what's needed to hand it over.
> 
> Ian

@Ian-Zz I had trouble in the past trying to transfer expired domains, but I'm willing to pay for the renewal. please check your @hotmail email to where to send me a payment request. We can go from there.

**Comment by Ian-Zz** [comment]
I chatted with domain com yesterday and they are to send me a transfer link/code in three days. I told asheroto that I would forward it to him. And I still will. Could. But... I'd rather it go (the responsibilities) to more then one person. Would hate to have this shit happen all over again.

How do you "domain" by "committee"?.
F this one person stuff.

**Comment by Ian-Zz** [comment]
A shared email account?  Is that a thing?

**Comment by dequi** [comment]
According to the domain.com [knowledgebase ](https://www.domain.com/help/article/user-roles-in-new-account-manager) you can have multiple users as administrative contact but there is always a primary contact that holds the keys sort of speaking. The primary contact is the legal registered owner. The only way is to set up a legal entity and have legal agreements between the parties involved if you want shared ownership. This is the same for all registrars I know (have worked with)

**Comment by asheroto** [comment]
I planned on creating a shared account under the new domain registrar. Even if one person is responsible, allowing additional trusted individuals access is a best practice to prevent against some type of disaster situation where someone becomes unavailable.

**Comment by Ian-Zz** [comment]
eventghost.net transfer code from domain.com has been forwarded to asheroto.
Good luck ya'll. And Have Fun

**Comment by automationghost** [comment]
so you juggling the domain around what does this mean ? Whats the point ?

**Comment by Ian-Zz** [comment]
No juggling here. I handed it over. and am "walking away" so to speak. You all do as you see fit.

**Comment by asheroto** [comment]
@chip33 was super helpful and emailed me a 482 MB archive of the website from the Wayback Machine. After analyzing, it looks like I can restore the website itself and the documentation. The forum will require some work since I only having the generated HTML and not the underlying database. I may be only to recreate it as a read-only forum without any posts.

If Kevin or someone else is able to retrieve the database backup, I could recreate the forum and import the database, otherwise it'll just be read-only.

@automationghost - There are a few reasons for dealing with the site:
- Keeping the domain itself prevents scammers from registering it and making others believe they're downloading a legitimate program.
- URLs in the program's Help menu currently go nowhere, leading users to question the status of the program.
- There is a lot of useful documentation available on the website.
- There are a lot of plugins, tweaks, and information available in the forums. Lots of backlinks such as Reddit or other forums link to EventGhost.net's forums. Since links are dead, users are stuck in the dark with the program.
- The GitHub repo links to the site.

Even if nothing works and we simply just put a page up that says "EventGhost is retired", that'd be better than nothing. :-) But I think at the very least I can restore some of the site and documentation. Hopefully I'll be able to tweak the forums to make a read-only version. Since I only have the generated HTML I can't make it so people can reply or comment; even this will take some work to get up. The goal is still to have registrar account with at least two or more persons with account access to prevent this type of situation from happening again. More details to come in the next few days. Transfer is pending.

@jibblet I do not believe Kevin was the original developer but rather one of the lead contributors. The original developer/owner is not known to me. Kevin has said he has been unable to reach them for many years without response. I don't know the full story.

Website backstory - The former hosting company where the website was hosted went bankrupt and an unknown reason, the website was unable to be recovered. I don't know the full story.

The good news is that good things are coming and thanks to a collective effort some parts of the site may be restored. :-) More to come.

**Comment by DanEdens** [comment]
It's far from retired! I am actually going to finish a rust version (AI agents are super fun. the Architecture of EG is just so fricking cool lol), 
and I'll fix the python once turning off the GIL is out of beta

**Comment by atrugiel69** [comment]
> It's far from retired! I am actually going to finish a rust version (AI agents are super fun. I'm 150 commits deep. The Architecture of EG is just so fricking cool lol), and I'll fix the python once turning off the GIL is out of beta

Will this version be compatible with previous ones?

**Comment by dequi** [comment]
> [@chip33](https://github.com/chip33) was super helpful and emailed me a 482 MB archive of the website from the Wayback Machine. After analyzing, it looks like I can restore the website itself and the documentation. The forum will require some work since I only having the generated HTML and not the underlying database. I may be only to recreate it as a read-only forum without any reply/post ability.

@asheroto No need to go trough all that effort. I have done this exercise and and it is rather futile; I can send you a backup of the website, wiki and forum (database backup not html rebuilds) up until march 2020. Just let me sanitize the private messages and I can have it for you on Monday (I just need to find your email somewhere i guess). I was at the stage of trying to get in touch with Kevin to see if he had a later backup but he never responded to my messages, and I stopped using Eventghost and didn't continue trying to get things back on track. Glad Kevin responds now, hope to see if he really retrieves his other machine tough.

> [@jibblet](https://github.com/jibblet) I do not believe Kevin was the original developer but rather one of the lead contributors. The original developer/owner is not known to me. Kevin has said he has been unable to reach them for many years without response. I don't know the full story.

some history as I came to know it:

Bitmonster was the original author, in 2010 he stopped the project and @Ian-Zz and some other users took over, mostly @Pako2 did the development. Later on @Blackwind and @CarsonF migrated the project to Github where Blackwind and @topic2k did development. Due to some different visions of the future of the project between Blackwind and Ian, Blackwind ended his commitment with the project and Topic2k and Kevin continued. Later there was only Kevin left; I don't know why Topic2k stopped contributing. Then when the server went down Kevin disappeared as well. 

As far as I know Ian was owner of the domain, Kevin did the hosting (phpbb/mediawiki) and Blackwind had the Eventghost Slack.
not sure who had access to all the different gmail email addresses. 

@blackwind do you still have access to the Slack and are you perhaps willing to share access?

**Comment by automationghost** [comment]
> > It's far from retired! I am actually going to finish a rust version (AI agents are super fun. I'm 150 commits deep. The Architecture of EG is just so fricking cool lol), and I'll fix the python once turning off the GIL is out of beta
> 
> Will this version be compatible with previous ones?


 I believe Kgschlosser made a python 3 version years ago for personal use which i think is what he is using. That means its not compatible ( since released version of  Eventghost running on python 2.

That means you will have a new(old) eventghost running on python 3 and all plugins need to be recreated.

**Comment by dequi** [comment]
@asheroto 
> I just need to find your email somewhere i guess

Found it in the other thread. Data backups I have coming your way shortly.

There also is an https://eventghost.readthedocs.io out there. I think it was set up by @kdschlosser so you need to ask him for the specifics.

**Comment by asheroto** [comment]
Thanks guys, good info. @dequi I will await your backup. I think what you mentioned sounds like the best choice to me, too.

Domain transfer is initiated, ETA 6 days or sooner per the registrar. That's just for the domain. I'll get the DNS, hosting, forum, and everything else prepared in the meantime.

We can determine the next steps in the coming days/weeks. Sounds like there are some changes and variants worth mentioning on the site or repo. We'll get it figured out. :-) I'll let you know when the dust has settled.

**Comment by DanEdens** [comment]
@chip33 Did i ever send you the darkmode version? 

Soon I'm going to add proper themeing, and start working on the actual repo. 
But I just hacked it in there lol. Sometimes it turns yellow and you have to reload. 
https://github.com/DanEdens/EventGhost

**Comment by chip33** [comment]
Hi Dan,
no you didn't send it to me.
I tried to copy what you did but it did not work so I just left it as is.
Yours,
Chris

**Comment by dequi** [comment]
@asheroto all data backups I have of the forum (until March 2020 - but after the forum attachment data loss of 2019), have been sent your way.

**Comment by asheroto** [comment]
The domain is now active and working! w00t!

https://eventghost.net

Thanks to @dequi we have a seemingly full-site backup to restore. There may be _some_ items missing but it looks like the majority is there. Details to come.

**Comment by dequi** [comment]
> There may be _some_ items missing but it looks like the majority is there

Well it is missing approximately 3 years of data: March 2020 -> December 2022. I'm hopefull @kdschlosser finds the time soon to check if he has a more recent backup.

**Comment by Ian-Zz** [comment]
((((https://eventghost.net/))))
Welcome Back. We have missed you so.

**Comment by WojciechNowakowski** [comment]
Hello guys. I am watching this thread since 2 weeks and it is great news to hear that EventGhost's page was restored. Kudos for you.
If there is some topic that an Embedded developer could help you with then feel free to reach me :)

**Comment by asheroto** [comment]
Homepage is restored. Docs are up. I've updated some of the links.

The forum and wiki may be a lost cause. The backup is old -- old enough where the underlying source code is built on a platform that doesn't support the current version of PHP. Rather than use an old version, for security reasons, we'll have to try and bring the existing data up-to-date with PHP 8. Unknown at this time if it will work but will keep everyone posted.

**Comment by reactsaas** [comment]
any plans to continue development on eventghost ? or is this simply a comeback for maintaining the website ?

**Comment by asheroto** [comment]
There have been some recent commits, I believe the other developers are working on its development.

https://github.com/EventGhost/EventGhost/commits/master/

Some of the other devs have been involved for a long time. I'm tasked with maintaining the website.

**Comment by molitar** [comment]
Thanks for the return of Eventghost!  I was having so many issues with the mediplayer plugin where it would not detect the application running until I restarted it and the OSD for volume was no longer showing. I am glad to say the new release fixed these bugs and glad to hear the website is returning!

**Comment by siparker** [comment]
> Homepage is restored. Docs are up. I've updated some of the links.
> 
> The forum and wiki may be a lost cause. The backup is old -- old enough where the underlying source code is built on a platform that doesn't support the current version of PHP. Rather than use an old version, for security reasons, we'll have to try and bring the existing data up-to-date with PHP 8. Unknown at this time if it will work but will keep everyone posted.

I am fairly good with this kind of thing if you need any assistance just drop me a line.

**Comment by molitar** [comment]
yeah be nice if the scripts from the old forum could be saved it had so much useful ones from there. If siparker can help save it than please contact him.

**Comment by asheroto** [comment]
@siparker can you contact me here?

https://asheroto.com

I have the info but haven't had any time to go through it. Perhaps we could collaborate to restore forum functionality.

**Comment by IRQ-HxH** [comment]
> Homepage is restored. Docs are up. I've updated some of the links.
> 
> The forum and wiki may be a lost cause. The backup is old -- old enough where the underlying source code is built on a platform that doesn't support the current version of PHP. Rather than use an old version, for security reasons, we'll have to try and bring the existing data up-to-date with PHP 8. Unknown at this time if it will work but will keep everyone posted.

Thanks for your effort asheroto!

Hopefully it will be possible to get the forum and Wiki back up running, lots of lost information otherwise.

Are the old forum/wiki source/raw data available to download somehow? Then it would at least be possible to search them locally with various tools.

**Comment by asheroto** [comment]
> Are the old forum/wiki source/raw data available to download somehow? Then it would at least be possible to search them locally with various tools.

@siparker was working on this a few months ago, but he may have gotten busy. The issue is the PHP version used is so outdated it would be a security risk run it as is. Upgrading is possible, but unfortunately I don't have the time to commit to this.

I have a SQL export but the problem with that is that it's so large, it's difficult to search since it's everything.

Ultimately I think the upgrade path is the best bet, but it requires a lot of little fixes to make it work. It's not just a straight upgrade, so will likely take some time. If I have another volunteer who is willing to take a go at it, I can send over some more details. The volunteer would need experience in forums, wikis, SQL, and PHP.

Hope that helps.

**Comment by IRQ-HxH** [comment]
Thanks for clarifying [asheroto](https://github.com/asheroto)! Hopefully it can get restored and brought online again somehow, IIRC there was lots of knowledge in that forum.

**Comment by siparker** [comment]
I still have it on my todo list. sorry for the , really quite excessive, delay. i will dedicate some time to it asap.

**Comment by atrugiel69** [comment]
> I still have it on my todo list. sorry for the , really quite excessive, delay. i will dedicate some time to it asap.

Thanks for all you'll be able to do. Much appreciated.

**Comment by siparker** [comment]
Step 1 done. 

<img width="1920" height="893" alt="Image" src="https://github.com/user-attachments/assets/b1b0e25e-fd3c-41e0-ae6e-40d81c77f4f6" />

@asheroto The latest posts here appear to be from 2020 is that what we are expecting?

**Comment by asheroto** [comment]
@siparker and I have chatted. He's got phpBB working on PHP 8.1, and phpBB updated to the latest. Nice work!

# Announcement! 🎉
Over the next few weeks, we will be bringing the forum back online and **reopening it to the public**. Our goal is to preserve the existing URL structure so previously shared links across the web will begin working again.

We are also developing a process for applying to become a moderator and for former moderators or admins to regain access to their accounts. A sign-up form and formal process will be released soon. More details will be announced once everything is ready.

**Comment by IRQ-HxH** [comment]
Thats great news and kudos to @siparker for his work!

**Comment by siparker** [comment]
> Thats great news and kudos to [@siparker](https://github.com/siparker) for his work!

Eventghost is an amazing tool and really deserves to be saved. I'm happy I could be part of that effort.

**Comment by atrugiel69** [comment]
Awesome news!!! Thanks to both asheroto and siparker for all they're doing!

**Comment by IRQ-HxH** [comment]
@siparker any progress on the forum?

**Comment by siparker** [comment]
> [@siparker](https://github.com/siparker) any progress on the forum?

Hi, 
Apologies. i have been out of action for a while. I have sent the forums files and db across to asheroto so he should be able to get it back up now. 
I have a half finished wiki also which ill try and finish off over the weekend and pass across .

**Comment by asheroto** [comment]
Hi everyone,

I've got the forum back from @siparker and will be working to reactivate it. For security, all passwords will reset and all moderators/admins will convert to regular user accounts.

We are currently looking for community members who have the time and availability to help moderate the forums. Because moderation requires ongoing attention and involvement, it may not be a good fit for those with very limited free time or demanding work schedules.

If you are interested in becoming a moderator, or if you are a former EventGhost Forums moderator who would like to reclaim your account, please complete [this form](https://form.typeform.com/to/z0f6PE5x).

Thank you

**Comment by IRQ-HxH** [comment]
Good to hear @asheroto. Looking forward to Forum going live!

**Comment by asheroto** [comment]
Unfortunately, no one has completed the form, and we really need a few moderators. Otherwise I'll have to release the forum as read-only, which I suppose is better than nothing.

I'll give it another two weeks and then plan on going live.

**Comment by wysocki** [comment]
I've been a forum member and have used EG for many years. I'm retired and would be willing to be a moderator if it means the survival of the forum but I'm not really sure what that all entails! 

Anyway I tried to sign up on the form but got stuck in a loop. When asked "Have you ever moderated any forums before?" I replied NO but then it loops back to "Username you would like to use on the forum", and the loop continues.

**Comment by asheroto** [comment]
Sorry about that, try it now.

**Comment by IRQ-HxH** [comment]
Like @wysocki I'm willing to give it a try even if I never moderated any forums before and RL may delay involment at times. I do belive a public forum is critical to share information and especially to get new blood (and hopefully old blood as well) involved to keep this program alive - it's a to good program to let die and old is not equal bad, sometimes to the contrary since newer software is often to **damn bloated**!

I filled out the form @asheroto linked to and will do my best to help out moderating if we can get this going...

**Comment by asheroto** [comment]
Thanks everyone. My day job keeps me pretty busy, but this is still on my to do list. If there are any other web developers interested in helping, it'd be good to have another set of eyes on the project.

I see two people have signed up to be a moderator. We may start out with read-only access at the beginning until after web traffic is analyzed, but will keep you posted.

**Comment by asheroto** [comment]
# The EventGhost Forums Are Back

We're happy to announce that the EventGhost community forums at [https://www.eventghost.net/forum](https://www.eventghost.net/forum) are online again!

## What happened

When the EventGhost forums went offline, years of community knowledge went with them. The outage was raised in this issue, where it became clear how much the archive was missed. A database backup from March 2020 was eventually located, and [@asheroto](https://github.com/asheroto) and [@siparker](https://github.com/siparker) have restored it, migrated it to current forum software, and brought it back online.

## What you should know

- **The backup dates from March 2020.** Any posts, accounts, or changes made after that point are unfortunately not included. Some content, links, and dates may be outdated.
- **The forum is currently read-only.** You can browse and search the full archive of over a decade of EventGhost discussions, plugins, and troubleshooting knowledge.
- **Sign-in and registration are temporarily disabled** while we finish securing and reorganizing things. We're working to restore both.
- **Existing accounts are preserved.** When sign-in returns, all members will need to use the password reset function to set a new password, since all old passwords were invalidated as a security precaution.

## Why it matters

The forum archive contains years of knowledge about EventGhost - plugin development, event macros, hardware quirks, and solutions to problems that aren't documented anywhere else. Keeping it browsable and searchable preserves that resource for everyone.

Thanks for your patience, and welcome back!

**Comment by IRQ-HxH** [comment]
Great work @asherto and @siparker! It's a great step since GitHub's forum is subpar for this purpose and hopefully old and new people find their way back!.

I know there are still interest in EG but many have probably given up since the forum been offline so long so we can **all** help by spreading the word in/on various forums/platforms.
